### ISSUE TITLE
`vec` does not work

### ISSUE THREAD

**Issue by Keno** [JuliaArrays/StaticArrays.jl#116]
```
julia> vec(StaticArrays.SArray{(2, 2),Int64,2,4}([1 2; 3 4]))
ERROR: MethodError: convert(::Type{StaticArrays.SArray{(2, 2),Int64,2,4}}, ::StaticArrays.SArray{(2, 2),Int64,2,4}) is ambiguous. Candidates:
  convert(::Type{SA}, sa::StaticArrays.StaticArray) where SA<:StaticArrays.StaticArray in StaticArrays at /Users/kfischer/.julia/v0.6/StaticArrays/src/core.jl:69
  convert(::Type{SA}, a::AbstractArray) where SA<:StaticArrays.StaticArray in StaticArrays at /Users/kfischer/.julia/v0.6/StaticArrays/src/core.jl:72
  convert(::Type{SA}, x1) where SA<:StaticArrays.StaticArray in StaticArrays at /Users/kfischer/.julia/v0.6/StaticArrays/src/core.jl:66
  convert(::Type{T}, x::T) where T in Base at essentials.jl:29
Stacktrace:
 [1] _reshape(::StaticArrays.SArray{(2, 2),Int64,2,4}, ::Tuple{Int64}) at ./reshapedarray.jl:142
 [2] vec(::StaticArrays.SArray{(2, 2),Int64,2,4}) at ./abstractarraymath.jl:33
```

**Comment by c42f** [comment]
I'm having trouble reproducing this for some reason.  Which version of julia and StaticArrays are you using?

**Comment by andyferris** [comment]
That looks like a v0.6 stacktrace.

@Keno, *StaticArrays* has been broken on v0.6 (due to the new "hyperpureness" of `@generated` function generators), and I'm slowly fixing things up on the `julia-0.6` branch. You can follow my progress at #113 (help also appreciated :) ).

**Comment by andyferris** [comment]
This seems to be somewhat fixed on the `julia-0.6` branch, now. It returns a `ReshapedArray`. Maybe I'll fix that up.

**Comment by andyferris** [comment]
I got it to return a proper `StaticArray` on the `julia-0.6` branch, so I'll close this issue.
